base/src/types.rs

```diff
@@ -229,6 +217,7 @@ pub enum BuiltinType {
     Array,
     /// Type constructor for functions, `(->) a b : Type -> Type -> Type`
     Function,
+    Type,
```

Comment:
I'm also thinking it might make sense to put it on the main enum, seeing as it is core to our language. Still umming over `Function` too...